### ISSUE TITLE
Required property does not work for falsy values

### ISSUE THREAD

**Issue by jawallace** [scotthovestadt/schema-object#72]
```
var SO = new SchemaObject({
    val: { 
        type: Number, 
        required: true 
    }
});

var o = new SO({ val: 0 });

o.isErrors(); // true
```

I have a change prepared that checks explicitly for `undefined `rather than forcing the value to a boolean, I will submit it in a PR.

**Comment by scotthovestadt** [comment]
@jawallace Thanks for the PR. It's merged and I'll publish it to NPM as a non-breaking change (minor version increment) later today. The required property is a recent addition (originating from a PR) so I don't think fixing it here will break anyone's use-case.

**Comment by jawallace** [comment]
Sounds great, thanks!

**Comment by jawallace** [comment]
@scotthovestadt Were you able to publish this to npm yet?

**Comment by scotthovestadt** [comment]
@jawallace Published as 4.0.9
